tests/unit/dcc_comms/test_mqtt_dcc_comms.py

```diff
@@ -0,0 +1,420 @@
+                     mqtt_msg_attr=self.mqtt_msg_attr)
+
+        # Check uuid file get created or not
+        self.assertFalse(os.path.exists(self.uuid_file))
+
+        # Check connect_soc call has been made
+        mock_connect.assert_called()
+
+    def test_init_with_invalid_msg_attr(self):
+        """
+        Test initialisation with invalid messaging attributes
+        :return: None
+        """
+        # Pass invalid mqtt messaging attribute
+        self.assertRaises(TypeError, lambda: MqttDccComms(edge_system_name=self.edge_system.name, url=self.url,
+                                                          port=self.port, clean_session=self.clean_session,
+                                                          mqtt_msg_attr=""))
+
+    @mock.patch.object(Mqtt, '__init__')
+    def test_connect(self, mock_init):
+        """
+        Test MqttDccComms _connect method.
+        :param mock_init: Mocked MQTT init method
+        :return: None
+        """
+        mock_init.return_value = None
+
+        # Create MqttDCCComms object
+        mqtt_dcc_comms = MqttDccComms(edge_system_name=self.edge_system.name, url=self.url, port=self.port,
+                                      identity=self.identity, tls_conf=self.tls_conf, qos_details=self.qos_details,
+                                      client_id=self.client_id, clean_session=self.clean_session,
+                                      protocol=self.protocol, transport=self.transport, keep_alive=self.keep_alive,
+                                      mqtt_msg_attr=self.mqtt_msg_attr,
+                                      enable_authentication=self.enable_authentication,
+                                      conn_disconn_timeout=self.connection_disconnect_timeout)
+
+        # Validate Mqtt __init__ called with following parameters
+        mock_init.assert_called_with(self.url, self.port, self.identity, self.tls_conf,
+                                     self.qos_details, self.client_id, self.clean_session, mqtt_dcc_comms.userdata,
+                                     self.protocol, self.transport, self.keep_alive, self.enable_authentication,
+                                     self.connection_disconnect_timeout)
+
+    @mock.patch.object(Mqtt, 'disconnect')
+    def test_disconnect(self, mock_disconnect):
+        """
+        Test MqttDccComms _disconnect method
+        :param mock_disconnect: Mocked MQTT disconnect method
+        :return: None
+        """
+        # Call MqttDccComms _disconnect
+        self.client._disconnect()
+
+        # Check call made to the Mqtt disconnect method
+        mock_disconnect.assert_called()
+
+    @mock.patch.object(Mqtt, 'publish')
+    def test_send_with_msg_attr(self, mock_publish):
+        """
+        Test MqttDccComms publish method with messaging attributes
+        :param mock_publish: Mocked MQTT publish method
+        :return: None
+        """
+
+        # Create subscribe MqttMessagingAttributes
```

Comment:
Change Typo.  It should be "Create publish MqttMessagingAttributes"
Also, create it with only publish related params.  Subscribe related will not be needed.

This MqttMessagingAttributes ans self.mqtt_msg_attr's pub and sub topics should be different